### DISCUSSION TITLE
Error reading table config properties

### DISCUSSION THREAD

**Discussion by vidhav** [developmentseed/tipg#158] [Q&A]
When I try to pass a list of properties in my configuration:

```properties
TABLE_CONFIG__public_mytable__PROPERTIES=["name","type","status"]
```

I get the following error from Pydantic:

```shell
table_config.public_mytable.properties
  Input should be a valid list [type=list_type, input_value='["name","type","status"]', input_type=str]
```

I believe this is because Pydantic settings only parses JSON in the top-level ([as stated in the docs](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#parsing-environment-variable-values)):

> JSON is only parsed in top-level fields, if you need to parse JSON in sub-models, you will need to implement validators on those models.

In this case, the values are nested and validated with a `TypedDict` ([see the code](https://github.com/developmentseed/tipg/blob/main/tipg/settings.py#L45)). For this to work, the code may need to be refactored to be `BaseSettings` with a custom validator to handle the parsing...

**Comment by vincentsarago** [comment]
Hi @vidhav 

Sorry I moved this to `discussion` because I thought it was a issue with the env name, but it seems that you are right, there is something going on with Pydantic (and it's also not tested on our side)

**Reply by vincentsarago** [reply]
This was fixed in #161
